subsys/serialization/include/rp_log.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef RP_LOG_H_
+#define RP_LOG_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define RP_MODULE_PREFIX  _CONCAT(RP_, RP_LOG_MODULE)
+/*
+ * The following macros from nrfx_config control the log messages coming from
+ * a given module:
+ * - RP <module>_CONFIG_LOG_ENABLED enables the messages (when set to 1)
+ * - RP <module>_CONFIG_LOG_LEVEL specifies the severity level of the messages
+ *   that are to be output.
+ */
+#if !IS_ENABLED(_CONCAT(RP_MODULE_PREFIX, _CONFIG_LOG_ENABLED))
+#define RP_MODULE_CONFIG_LOG_LEVEL 0
+#else
+#define RP_MODULE_CONFIG_LOG_LEVEL \
+	_CONCAT(RP_MODULE_PREFIX, _CONFIG_LOG_LEVEL)
+#endif
+
+#if	RP_MODULE_CONFIG_LOG_LEVEL == 0
+#define RP_MODULE_LOG_LEVEL		LOG_LEVEL_NONE
+#elif	RP_MODULE_CONFIG_LOG_LEVEL == 1
+#define RP_MODULE_LOG_LEVEL		LOG_LEVEL_ERR
+#elif	RP_MODULE_CONFIG_LOG_LEVEL == 2
+#define RP_MODULE_LOG_LEVEL		LOG_LEVEL_WRN
+#elif	RP_MODULE_CONFIG_LOG_LEVEL == 3
+#define RP_MODULE_LOG_LEVEL		LOG_LEVEL_INF
+#elif	RP_MODULE_CONFIG_LOG_LEVEL == 4
+#define RP_MODULE_LOG_LEVEL		LOG_LEVEL_DBG
+#endif
+#include <logging/log.h>
+LOG_MODULE_REGISTER(RP_MODULE_PREFIX, RP_MODULE_LOG_LEVEL);
+
+/**
+ * @brief Macro for logging a message with the severity level ERR.
+ *
+ * @param ... printf-style format string, optionally followed by arguments
+ *            to be formatted and inserted in the resulting string.
+ */
+#define RP_LOG_ERR(...)  LOG_ERR(__VA_ARGS__)
```

Comment:
Why is it necessary for RP to have it's own logging wrappers?

Reply:
This is a forwarding to the zephyr log, since the rp_log.h is defined outside in nrfxlib serialization/template/rp_log.h
https://github.com/NordicPlayground/nrfxlib/pull/149/files#diff-bbb0e4fb4bc97291d92201967fdf6797R28

Further details on this here:
https://projecttools.nordicsemi.no/confluence/display/NCS/BLE+Host-level+split+on+nRF53+-+architecture

Reply:
But we are inventing rp_log.h, right? Could we not use the Zephyr API directly and then have non-zephyr platforms implement it?

I don't see why the non-zephyr platforms would prefer the rp_log API over the Zephyr API.

Reply:
I moved configuration to Kconfig. @SebastianBoe I am not exactly sure what you mean  ?

Reply:
I guess what @SebastianBoe meant is that RP logging macros are redundant. We can modify the template in nrfxlib (just remove ```RP_``` prefix from macros) to use Zephyr macros directly

https://github.com/NordicPlayground/nrfxlib/pull/149/files#diff-bbb0e4fb4bc97291d92201967fdf6797R28

In this case, RP logger abstraction in nrfxlib will match Zephyr logger API (via ```logging/log.h``` include in ```rp_log.h```), so macro overrides in this file won't be necessary (~65 fewer lines of code).

And still, if somebody would like to use RP Serialization libraries outside of Zephyr context, they will just need to override ```LOG_INF``` macro and other macros instead of ```RP_LOG_INF``` macros, so it doesn't really matter.

However, if we decide to simplify it this way, I can see a few drawbacks:
- a violation of naming convention that we use inside each module/library in the template file (no "RP_" prefix)
- no demonstration of how to use logging template (since the RP logging abstraction and Zephyr logging API is the same)

Reply:
Good point @kapi-no. Thank you for clarifying.